devicetypes/smartthings/smartsense-motion-sensor.src/smartsense-motion-sensor.groovy

```diff
@@ -183,7 +184,8 @@ private Map getBatteryResult(rawValue) {
 			// OR we have received the same reading twice in a row
 			// OR we don't currently have a battery reading
 			// OR the value we just received is at least 2 steps off from the last reported value
-			if(state?.lastVolts == null || state?.lastVolts == volts || device.currentState("battery")?.value == null || Math.abs(curValVolts - volts) > 0.1) {
+			// OR the device's firmware is older than 1.15.7
+			if(state?.lastVolts == null || state?.lastVolts == volts || device.currentState("battery")?.value == null || Math.abs(curValVolts - volts) > 0.1 || isFirmwareVerLessThan(1, 15, 7)) {
```

Comment:
Rather than setting `result = null` on line 196 I think we should generate the event with the current value (`device.currentState("battery").value`). This way we'll know that we're still getting battery reports even if the value isn't changing.

Reply:
Yeah, I can add that @tpmanley . I believe if the value is the same as the last event value the platform won't create/propagate it, so I will need to add result.isStateChange = true to make sure the event isn't dropped. It seems this has been affecting device health for some devices, as well.

Reply:
I suggest putting the firmware version check first since that will be the most common case